src/Cake.Scripting.Transport/Serialization/CakeScriptSerializer.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.IO;
+using Cake.Scripting.Abstractions.Models;
+
+namespace Cake.Scripting.Transport.Serialization
+{
+    internal static class CakeScriptSerializer
+    {
+        public static void Serialize(BinaryWriter writer, CakeScript script)
+        {
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+            if (script == null)
+            {
+                throw new ArgumentNullException(nameof(script));
+            }
+
```

Comment:
Perhaps write an identifier for the type being serialized as well as a version (`byte` or `ushort`)?